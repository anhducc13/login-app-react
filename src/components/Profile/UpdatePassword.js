import React, { useState } from 'react';
import { Form, Input, Typography, Button } from 'antd';
import { authServices } from 'services';
import { toast } from 'react-toastify';
import { RULES_PASSWORD } from 'constants/RuleValidators';

const { Title } = Typography;
const { Password } = Input;

const UpdatePassword = (props) => {
  const formItemLayout = {
    labelCol: {
      xs: { span: 24 },
      sm: { span: 6 },
    },
    wrapperCol: {
      xs: { span: 24 },
      sm: { span: 12 },
      lg: { span: 8 },
    },
  };

  const tailFormItemLayout = {
    wrapperCol: {
      xs: {
        span: 24,
        offset: 0,
      },
      sm: {
        span: 12,
        offset: 6,
      },
      lg: {
        span: 8,
        offset: 6,
      },
    },
  };

  const [loading, setLoading] = useState(false);

  const { getFieldDecorator, getFieldValue, validateFields, resetFields } = props.form;

  const handleSubmit = (e) => {
    e.preventDefault();
    validateFields((error, values) => {
      if (!error) {
        setLoading(true);
        const params = {
          "old_password": values.password,
          "new_password": values.newPassword,
        }
        authServices.updatePasswordUser(params)
          .then(() => {
            setLoading(false);
            toast.success(`Update password success!`);
            props.history.push('/');
          })
          .catch(err => {
            setLoading(false);
            switch (err) {
              case 400:
                resetFields();
                break;
              case 403:
                props.history.push('/403');
                break;
              case 404:
                props.history.push('/404');
                break;
              default:
                props.history.push('/500');
            }
          })
      }
    })
  }

  return (
    <div style={{
      backgroundColor: '#FFFFFF',
      borderRadius: 5,
      padding: 16
    }}
    >
      <Title
        level={2}
        style={{
          marginBottom: 20
        }}
      >
        Update Password
      </Title>
      <Form {...formItemLayout}>
        <Form.Item label="Password">
          {getFieldDecorator('password', {
            rules: RULES_PASSWORD,
            validateFirst: true,
            validateTrigger: null,
          })(
            <Password />
          )}
        </Form.Item>
        <Form.Item label="New Password">
          {getFieldDecorator('newPassword', {
            rules: RULES_PASSWORD,
            validateFirst: true,
            validateTrigger: null,
          })(
            <Password />
          )}
        </Form.Item>
        <Form.Item label="Confirm">
          {getFieldDecorator('confirm', {
            rules: [
              {
                required: true,
                message: "Required"
              },
              {
                pattern: new RegExp(getFieldValue('newPassword')),
                message: "Confirm password not match"
              }
            ],
            validateFirst: true,
            validateTrigger: null,
          })(
            <Password />
          )}
        </Form.Item>
        <Form.Item {...tailFormItemLayout}>
          <Button
            type="primary"
            htmlType="submit"
            block
            icon="save"
            loading={loading}
            onClick={handleSubmit}
          >
            Save
          </Button>
        </Form.Item>
      </Form>
    </div>
  )
}

export default Form.create()(UpdatePassword);