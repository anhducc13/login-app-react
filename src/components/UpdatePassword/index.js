import React from 'react';

export default (props) => {
  return (
    <div>
      sawdgrgwrg
    </div>
  )
}
