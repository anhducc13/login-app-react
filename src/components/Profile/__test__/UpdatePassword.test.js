import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import UpdatePassword from '../UpdatePassword';

describe('Update Password Page Render', () => {
  it('should render correctly', () => {
    const component = shallow(<UpdatePassword />);
    expect(component).toMatchSnapshot();
  });
});

describe('Update Password form validate', () => {
  it('validate test case 1', () => {
    const component = mount(
      <MemoryRouter>
        <UpdatePassword />
      </MemoryRouter>
    );
    component.find('input[id="password"]').simulate('change', { target: { value: "" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Required")
    component.unmount()
  })

  it('validate test case 2', () => {
    const component = mount(
      <MemoryRouter>
        <UpdatePassword />
      </MemoryRouter>
    );
    component.find('input[id="password"]').simulate('change', { target: { value: "ab" } });
    component.find('input[id="newPassword"]').simulate('change', { target: { value: "anhducc14" } });
    component.find('input[id="confirm"]').simulate('change', { target: { value: "a" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').length)
      .toBe(3)
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Content upper letter, lower letter and number, no special character")
    component.unmount()
  })

  it('validate test case 3', () => {
    const component = mount(
      <MemoryRouter>
        <UpdatePassword />
      </MemoryRouter>
    );
    component.find('input[id="password"]').simulate('change', { target: { value: "Anhducc14" } });
    component.find('input[id="newPassword"]').simulate('change', { target: { value: "Anhducc15" } });
    component.find('input[id="confirm"]').simulate('change', { target: { value: "a" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').length)
      .toBe(1)
    expect(component.find('div.ant-form-explain').first().text())
      .toBe("Confirm password not match")
    component.unmount()
  })

  it('validate test case 4', () => {
    const component = mount(
      <MemoryRouter>
        <UpdatePassword />
      </MemoryRouter>
    );
    component.find('input[id="password"]').simulate('change', { target: { value: "Anhducc14" } });
    component.find('input[id="newPassword"]').simulate('change', { target: { value: "Anhducc15" } });
    component.find('input[id="confirm"]').simulate('change', { target: { value: "Anhducc15" } });
    component.find('button[type="submit"]').simulate('click');
    expect(component.find('div.ant-form-explain').length)
      .toBe(0)
    component.unmount()
  })
});