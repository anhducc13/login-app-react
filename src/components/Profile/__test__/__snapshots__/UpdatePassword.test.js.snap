// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Password Page Render should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "padding": 16,
    }
  }
>
  <Title
    level={2}
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    Update Password
  </Title>
  <Form
    colon={true}
    hideRequiredMark={false}
    labelCol={
      Object {
        "sm": Object {
          "span": 6,
        },
        "xs": Object {
          "span": 24,
        },
      }
    }
    layout="horizontal"
    onSubmit={[Function]}
    wrapperCol={
      Object {
        "lg": Object {
          "span": 8,
        },
        "sm": Object {
          "span": 12,
        },
        "xs": Object {
          "span": 24,
        },
      }
    }
  >
    <FormItem
      hasFeedback={false}
      help=""
      label="Password"
      validateStatus=""
    >
      <Password
        action="click"
        inputPrefixCls="ant-input"
        name="password"
        onChange={[Function]}
        prefixCls="ant-input-password"
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
      help=""
      label="New Password"
      validateStatus=""
    >
      <Password
        action="click"
        inputPrefixCls="ant-input"
        name="new-password"
        onChange={[Function]}
        prefixCls="ant-input-password"
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
      help=""
      label="Repeat New Password"
      validateStatus=""
    >
      <Password
        action="click"
        inputPrefixCls="ant-input"
        name="repeat-password"
        onChange={[Function]}
        prefixCls="ant-input-password"
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
      wrapperCol={
        Object {
          "lg": Object {
            "offset": 6,
            "span": 8,
          },
          "sm": Object {
            "offset": 6,
            "span": 12,
          },
          "xs": Object {
            "offset": 0,
            "span": 24,
          },
        }
      }
    >
      <Button
        block={true}
        ghost={false}
        htmlType="submit"
        icon="save"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Save
      </Button>
    </FormItem>
  </Form>
</div>
`;
